coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,313 @@
+                
+            elif project.startswith(ICA_PREFIX):
+                if project_status == 'Inactive' and end_date is not None:
+                    self.stdout.write(self.style.ERROR('{} {} '
+                    'for inactive ICA project {} has an end date '
+                    '(it shouldn\'t as it\'s inactive).'
+                        .format(resource, id, project)))
+                if project_status == 'Active' \
+                        and not any(end_date != ica.end_date \
+                            for ica in ICA_ALLOCATION_PERIODS):
+                    self.stdout.write(self.style.ERROR('{} {} '
+                    'for {} ICA project {} has an end date '
+                    'that is different from all ICA allocation periods.' \
+                        .format(resource, id, project_status.lower(), project)))
+            elif project.startswith(RECHARGE_CONDO_PREFIX):
+                if end_date is not None:
+                    self.stdout.write(self.style.ERROR('{} {} '
+                    'for {} Recharge or Condo allocation project {} has '
+                    'an end date. (it shouldn\'t' \
+                        .format(id, project_status.lower(), project)))
+    
+    def handle_project_inactive(self):
+        '''
+        Assert that inactive Projects have "Expired" allocations and zero SUs.
+        '''
+        inactive_projects = Project.objects \
+                                .prefetch_related('allocation_set__status',
+                                    'allocation_set__resources',
+                                    'allocation_set__allocationattribute_set') \
+                                .filter(status__name='Inactive') \
+                                .order_by('name')
+        
+        for project in inactive_projects:
+            for allocation in project.allocation_set \
+                              .select_related('status') \
+                              .prefetch_related('allocationattribute_set',
+                                                'resources'):
+                resource = allocation.resources.first().name
+                style = (lambda x:
+                        self.style.WARNING('CURRENTLY EXPECTED BEHAVIOR: ') \
+                        + self.style.WARNING(x)) if \
+                        resource.endswith('Directory') \
+                        else self.style.ERROR
+                if allocation.status.name != 'Expired':
+                    self.stdout.write(style('Project {} ({})'
+                    ' is inactive and has an unexpired {} allocation {}.' \
+                        .format(project.id, project.name,
+                                resource, allocation.id)))
+                try:
+                    allocation_attribute = allocation.allocationattribute_set \
+                        .get(allocation_attribute_type__name='Service Units')
+                    if allocation_attribute.value == 0:
```

Comment:
The value is a string, so I don't think this case would ever fire.

Reply:
Also, should this be `!=`?

Reply:
Yeah, it should be != oops. I found 2 different 0 values for the attribute, "0" and "0.00". Are there any other ones I should be aware of (like None or "0.0")

Reply:
We typically try to coerce it into a `Decimal`--if that fails, then there's a corrupted value.